### PR TITLE
ARROW-7503: [Rust] [Parquet] Fix build failures

### PR DESCRIPTION
**Pull request by nevi-me**
This replaces the deprecated `Error::description()` with a simpler ToString. I'm not familiar with the `quick-error` crate, but did my best from what I could see on the https://github.com/tailhook/quick-error/commit/b26770f47b34e5be08ecef706e82e2e474074e9e commit that addressed the deprecation warning/error.

### CODE REVIEW

**Comment by nevi-me** [comment]
PTAL @liurenjie1024

File: rust/parquet/src/errors.rs [hunk 1]
```diff
@@ -33,7 +32,7 @@ quick_error! {
       /// Returned when code violates normal workflow of working with Parquet files.
       General(message: String) {
           display("Parquet error: {}", message)
-              description(message)
+              display(message)
```
Comment (liurenjie1024) [inline rust/parquet/src/errors.rs]: We already have a display statement? Why another one?

File: rust/parquet/src/errors.rs [hunk 2]
```diff
@@ -44,25 +43,25 @@ quick_error! {
       /// Returned when functionality is not yet available.
       NYI(message: String) {
           display("NYI: {}", message)
-              description(message)
+              display(message)
```
Comment (liurenjie1024) [inline rust/parquet/src/errors.rs]: Remove this?

File: rust/parquet/src/column/writer.rs [hunk 3]
```diff
@@ -892,8 +890,8 @@ mod tests {
         assert!(res.is_err());
         if let Err(err) = res {
             assert_eq!(
-                err.description(),
-                "Inconsistent length of definition and repetition levels: 3 != 2"
+                err.to_string(),
```
Comment (liurenjie1024) [inline rust/parquet/src/column/writer.rs]: In fact to_string is derived from Display trait. I would suggest to replace `to_string` with `display`. Though the results are same, but `display` matches with rust convention better.

File: rust/parquet/src/errors.rs [hunk 4]
```diff
@@ -44,25 +43,25 @@ quick_error! {
...
       }
       /// "End of file" Parquet error.
       /// Returned when IO related failures occur, e.g. when there are not enough bytes to
       /// decode.
       EOF(message: String) {
           display("EOF: {}", message)
-              description(message)
+              display(message)
```
Comment (liurenjie1024) [inline rust/parquet/src/errors.rs]: Remove this?

File: rust/parquet/src/errors.rs [hunk 5]
```diff
@@ -44,25 +43,25 @@ quick_error! {
...
       }
       /// Arrow error.
       /// Returned when reading into arrow or writing from arrow.
       ArrowError(message:  String) {
           display("Arrow: {}", message)
-              description(message)
+              display(message)
```
Comment (liurenjie1024) [inline rust/parquet/src/errors.rs]: Same as above.

File: rust/parquet/src/errors.rs [hunk 6]
```diff
@@ -44,25 +43,25 @@ quick_error! {
...
               from(e: ArrowError) -> (format!("underlying Arrow error: {:?}", e))
       }
       IndexOutOfBound(index: usize, bound: usize) {
           display("Index {} out of bound: {}", index, bound)
-              description("Index out of bound error")
+              display("Index out of bound error")
```
Comment (liurenjie1024) [inline rust/parquet/src/errors.rs]: Same as above.

**Review by liurenjie1024 — CHANGES_REQUESTED** [review]
Thanks @nevi-me for this PR. Left some comments.

File: rust/parquet/src/column/writer.rs (hunk 3 above)
Comment (nevi-me) [reply rust/parquet/src/column/writer.rs]: `err.display()` doesn't work, that's why I ended up resorting to `err.to_string()`. Would you be able to address the build failure instead? I can close this PR. 

As I mentioned in the PR, this is my first time coming across `quick-error` so I was bound to not do things correctly.

Comment (liurenjie1024) [reply rust/parquet/src/column/writer.rs]: Sorry, you should use `format!("{}", err)`. I took a look at the build failures mentioned in jira and I think your changes are good enough to resolve it.

File: rust/parquet/src/errors.rs (hunk 1 above)
Comment (nevi-me) [reply rust/parquet/src/errors.rs]: My mistake, was doing this very late at night. I've removed the duplicates

File: rust/parquet/src/column/writer.rs (hunk 3 above)
Comment (nevi-me) [reply rust/parquet/src/column/writer.rs]: I've used `format!()` now

**Comment by liurenjie1024** [comment]
Thanks @nevi-me . LGTM

**Pull request closed without merging**